server/src/middleware/auth.js

```diff
@@ -0,0 +1,20 @@
+const jwt = require("jsonwebtoken");
+const config = require("../config");
+
+module.exports = {
+    async loginRequired(req,res,next) {
+        try{
+            const token = req.headers.authorization.split(" ")[1];
+            jwt.verify(token, config.privateKey, (err, decoded) => {
+                const d = new Date();
```

Comment:
If possible, please include an error check as well.

Reply:
How do you want the err to be handled? Should I pass it onto the next function or return right away?